app/src/main/java/org/p2p/wallet/history/ui/history/HistoryFragment.kt

```diff
@@ -53,10 +62,18 @@ class HistoryFragment :
     }
 
     override fun showPagingState(state: PagingState) {
+        isRefreshing = state is PagingState.Loading && state.isRefresh
         adapter.setPagingState(state)
-        binding.shimmerView.isVisible = state == PagingState.InitialLoading
-        binding.refreshLayout.isRefreshing = state is PagingState.Loading && state.isRefresh
-        binding.refreshLayout.isVisible = state != PagingState.InitialLoading
+        with(binding) {
+            shimmerView.isVisible = state == PagingState.InitialLoading
+            refreshLayout.isRefreshing = state is PagingState.Loading && state.isRefresh
+            refreshLayout.isVisible = state != PagingState.InitialLoading
+            historyRecyclerView.alpha = if (isRefreshing) {
```

Comment:
I think it's not the best way. 
Would be better to add a fullscreen view with half-transparent background and make it clickable to prevent clicks. 
And just show it when it's refreshing. 

Otherwise it's only simulating instead of real behavior. 
It maybe broken if some accidentally delete checking for `isRefreshing` and showing the real blocking view is cheap as well

Reply:
yes, I agree, I can make it it was a second way for implementation 

ok will do it

Reply:
we can create something like Fullscreen progress view

Reply:
?